ps_socialfollow.php

```diff
@@ -187,11 +202,32 @@ public function getConfigFieldsValues()
     {
         $result = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $result['blocksocial_' . $social] = Configuration::get('BLOCKSOCIAL_' . strtoupper($social));
+            if (!empty(Configuration::get('BLOCKSOCIAL_' . strtoupper($social)))) {
+                $this->upgradeConfiguration('BLOCKSOCIAL_' . strtoupper($social));
+            }
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $conf = Configuration::get('BLOCKSOCIAL_' . strtoupper($social), $id_lang);
```

Comment:
Nice-to-have: Could we find a way to gather all this `'BLOCKSOCIAL_' . strtoupper($social)` logic into a single method so we stop having concatenations everywhere?

Maybe `function buildBlockIdentifier($socialLink)`

Reply:
In this function it's used 3 times so it would be good to be able not to repeat it

Reply:
... or a local variable